corehq/apps/domain/models.py

```diff
@@ -53,7 +53,6 @@
 
 from corehq.apps.app_manager.const import AMPLIFIES_NO, AMPLIFIES_NOT_SET, AMPLIFIES_YES
 
-from .project_access.models import SuperuserProjectEntryRecord  # noqa
```

Comment:
The `noqa` caught my eye, so I blamed this, and this import was added in a commit that doesn't reference `SuperuserProjectEntryRecord` at all: https://github.com/dimagi/commcare-hq/commit/cca95dfa5b8617238c1772b01e44d5f209b6d3ae#diff-f20e46651cf77f677565a2f0fb741b63

It looks deliberate. @dannyroberts any insight?

Reply:
What do you mean?  The commit you linked shows the import being added, as well as the model being added.  My guess is since `corehq.apps.domain.project_access` isn't a "django app", the model is imported here so automatic discovery can happen.

Reply:
Yeah looking at that commit, my take is that while `corehq.apps.domain` is a django app, `corehq.apps.domain.project_access` is _not_, so in order to have this model get picked up by django, I have to import it from the location in which it was declared into a django app's `models.py` file.

Not sure how much I want to stand behind that pattern; I'm guessing my thinking was that I wanted to separate out the stuff that had to do with this new behind-the-scenes feature but didn't think it deserved its own django app.

But anyway, given that I did what I did, I agree removing this import will/could break things by not unregistering `SuperuserProjectEntryRecord` as a django model.

Reply:
> the model is imported here so automatic discovery can happen

That's the part I'm missing - what's automatic discovery? (can offline if easier)

Reply:
>  commit that doesn't reference SuperuserProjectEntryRecord at all

Sorry for confusion, I mis-typed this: I meant that this *file* doesn't reference SuperuserProjectEntryRecord even in the commit where the import was added.

Reply:
> what's automatic discovery?

I have only a vague understanding of it, but I believe the gist of it is that in order for Django to be aware of these models (eg: for the migrations framework), the models must be defined in a `models.py` file inside an app declared in `settings.INSTALLED_APPS`

Reply:
Just looked it up:
> You must define or import all models in your application’s models.py or models/__init__.py. Otherwise, the application registry may not be fully populated at this point, which could cause the ORM to malfunction.

> Once this stage completes, APIs that operate on models such as get_model() become usable.

https://docs.djangoproject.com/en/2.2/ref/applications/